apps/restconf/restconf_methods.c

```diff
@@ -653,120 +621,223 @@ int get_xval(
         cxobj *xn = vec[j];
         tmp_val = xml_body(xn);
     }
-    cpy_str(val, tmp_val, TEMP_STR_MALLOC_SIZE);
+    cbuf_append_str(val, tmp_val);
     return 0;
 }
 
-int do_replace (
+// TODO - add this to cligen repo if it is approved
+/*! Truncate a cbuf
+  *
+  * @param [in]  cb  cligen buffer allocated by cbuf_new(), may be reallocated.
+  * @param [in]  int pos position at which to truncate
+  * @retval new cbuf containing the truncated string (old buffer remains as it was)
+  * @retval NULL   Error
+  */
+cbuf*
+cbuf_trunc(cbuf       *cb,
+		int       pos)
+{
+    if (pos < 0 || pos > cb->cb_strlen){
+	errno = EINVAL;
+	return NULL;
+    }
+    /* Ensure buffer is right size */
+    cbuf* new_buf = cbuf_new_alloc(pos + 1);
+    if (new_buf == NULL)
+        return NULL;
+    strncpy(new_buf->cb_buffer, cb->cb_buffer, pos);
+    new_buf->cb_strlen = pos;
+    return new_buf;
+}
+
+/*! Add square brackets after the surrounding curly brackets in JSON
+  * Needed, in order to modify the result of xml2json_cbuf() to be valid input
+  * to api_dta_post() and api_dta_write()
+  * @param [in]  x_simple_patch  a cxobj to pass to xml2json_cbuf()
+  * @retval new cbuf with the modified json
+  * @retval NULL   Error
+  */
+
+static cbuf* yang_patch_xml2json_modified_cbuf(cxobj* x_simple_patch)
+{
+    cbuf *json_simple_patch = cbuf_new();
+    if (json_simple_patch == NULL)
+        return NULL;
+    cbuf* cb = cbuf_new();
```

Comment:
Clixon style: please move variable declarations to top of function, ie cb, json_simple_patch_tmp, brace_count.
Probably more cases.
Its OK to add them in a new body {}, but I get confused when they appear after code lines.
I know it is OK in current C, but my personal view is it is difficult to see what appears on the stack.